components/kyma-environment-broker/internal/monitoring/monitoring.go

```diff
@@ -102,25 +103,14 @@ func (c *client) IsPresent(releaseName string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	listAction := action.NewList(cfg)
-	listAction.Deployed = true
-	listAction.Superseded = true
-	listAction.Uninstalling = true
-	listAction.Deployed = true
-	listAction.Failed = true
-	listAction.Pending = true
-
-	releases, err := listAction.Run()
-	if err != nil {
-		return false, kebError.AsTemporaryError(err, "unable to list releases")
+	histClient := action.NewHistory(cfg)
+	histClient.Max = 1
+	if _, err := histClient.Run(releaseName); err == driver.ErrReleaseNotFound {
+		return false, nil
+	} else if err != nil {
+		return false, kebError.AsTemporaryError(err, "unable to get helm history")
 	}
-	for _, rel := range releases {
-		if rel.Name == releaseName {
-			return true, nil
-		}
-	}
-
-	return false, nil
+	return true, nil
```

Comment:
You should also reimplement `IsDeployed` with a similar logic. After getting the helm release with the history action `.Run()`, you can check the current state of the release.

Reply:
IsDeployed also fixed